sqlmesh/core/loader.py

```diff
@@ -461,23 +500,46 @@ def _load_python_models(
                 new = registry.keys() - registered
                 registered |= new
                 for name in new:
-                    model = registry[name].model(
-                        path=path,
-                        module_path=self.config_path,
-                        defaults=self.config.model_defaults.dict(),
-                        macros=macros,
-                        jinja_macros=jinja_macros,
-                        dialect=self.config.model_defaults.dialect,
-                        time_column_format=self.config.time_column_format,
-                        physical_schema_mapping=self.config.physical_schema_mapping,
-                        project=self.config.project,
-                        default_catalog=self.context.default_catalog,
-                        variables=self._variables,
-                        infer_names=self.config.model_naming.infer_names,
-                        audit_definitions=audits,
-                    )
-                    if model.enabled:
-                        models[model.fqn] = model
+                    registered_entrypoint = registry[name]
+
+                    gateway = registered_entrypoint.kwargs.get("gateway") or ""
+                    blueprints = registered_entrypoint.kwargs.pop("blueprints", None)
+
+                    for blueprint in blueprints or [None]:
```

Comment:
We need to consolidate the logic that parses blueprints with the one above.

Reply:
Done, both SQL and Python model blueprinting now happens within `create_model_blueprints`.